OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -110,10 +110,13 @@ public List<CPos> FindUnitPathToRange(CPos source, SubCell srcSub, WPos target,
 			// Correct for SubCell offset
 			target -= world.Map.Grid.OffsetOfSubCell(srcSub);
 
+			var rangeLengthSquared = range.LengthSquared;
```

Comment:
A `// PERF: avoid recalculating this multiple times in the LINQ query.` might be helpful here?

Reply:
Yeah, but it is also too obvious, should be common practice.

These recalculations in LINQ code tend to slip in. Also because operator i.e. +/- overloading leaves the impression that some operations are cheap.